compat/mingw.c

```diff
@@ -3396,8 +3396,7 @@ int msc_startup(int argc, wchar_t **w_argv, wchar_t **w_env)
 	 */
 	ALLOC_ARRAY(my_utf8_argv, argc + 1);
 	ALLOC_ARRAY(save, argc + 1);
-	save[0] = my_utf8_argv[0] = wcstoutfdup_startup(buffer, _wpgmptr, maxlen);
```

Comment:
I am afraid that we really need the `argv[0]` as before, since we deduce the relative location of the `libexec/git-core/` directory from the absolute path of the executable. With the proposed change, we would lose that ability as we may very well get just the bare name of the `.exe` file, without any path.

Reply:
Just now, before I saw this comment, I ran into a crash because of it at [exec_cmd.c#L20](https://github.com/git/git/blob/v2.16.2/exec_cmd.c#L20), when I did:
```
C:\git\bug\test>C:\PortableGit\mingw32\libexec\git-core\git-status.exe
```

I see that it needs [git_extract_argv0_path()](https://github.com/git/git/blob/v2.16.2/exec_cmd.c#L31) to set up `argv0_path`, which was still NULL causing the crash.

It seems wrong to me to take the executable path and put it in argv[0], to later use argv[0] to calculate the system prefix. If we want to know our system prefix relative to our executable, then use the executable path directly instead of using argv[0].

Reply:
I guess we should piggy-back onto [the patches](https://github.com/danjacques/git/commits/runtime-prefix) provided by @danjacques to make the `RUNTIME_PREFIX` business less Windows-specific. We would then provide simply another function to determine the absolute path of the current executable and add it here: https://github.com/danjacques/git/commit/df3ddf76f117243790f2506caa2d673ba7be6c22#diff-9fb30e33d515122efff4fa404ee3a2d9R170

However, those patches have been lying dormant for a while and are *far* from our `master`. But if you can help that effort, I think we can integrate them into Git for Windows' `master` as soon as they are stable enough to hit core Git's `next`.

Reply:
Any idea what effort remains for getting `exec_cmd: RUNTIME_PREFIX on some POSIX systems` stable enough? I was refactoring in a similar direction (kgybels/git@dca5afd0342e241a631b793c937f51497b163a20), however, @danjacques's work is much more evolved.

Reply:
@danjacques' work is much more involved, I agree. The hold-up was some clash with @avar's Perl changes, and now that they made it in, @danjacques mentioned he'd work on his patches again.

Since they will probably make it into core Git, I'd rather work together with @danjacques than resolve merge conflicts later ;-)

The patches should rebase pretty cleanly with the only exception of the Perl changes, which should no longer be necessary IIUC.

Would you mind doing this and adding your changes on top?